pymatgen/io/lammps/inputs.py

```diff
@@ -33,10 +33,114 @@
+            is_stage_header: Set this to True, if the comment should be
+                treated as stage/block header to be used at the beginning of
+                each stage/block. Headers are preceded by '#' (single hash)
+                in the generated input.
+        """
+
+        if is_stage_header:
+            self._add_command("header", "# " + comment)
+        else:
+            self.ncomments += 1
+            self._add_command(f"comment_{self.ncomments}", "## " + comment)
+
+    def add_commands(self, commands: Union[str, list, dict], stage_name: str = None):
+        """
+        Adds LAMMPS command/s and its arguments to LAMMPS input file.
+
+        Args:
+            commands: LAMMPS command/s. Can pass single string or list of LAMMPS command
+            with its arguments. Also accepts dictionary of LAMMPS commands are
+            corresponding arguments as key value pairs.
+            stage_name: If a stage name is mentioned, the command is added
+                under that stage block else latest stage is assumed.
+        """
+
+        if self.curr_stage_name is None:
+            self.init_stage()
+        self._add_stage_name(stage_name)
+
+        if isinstance(commands, str):
+            self._add_command(commands)
+        elif isinstance(commands, list):
+            for command in commands:
+                self._add_command(command=command)
+        elif isinstance(commands, dict):
+            for command, args in commands.items():
+                self._add_command(command=command, args=args)
+
+    def add_stage(self, stage_commands: Union[list, dict], header: str = None, description: str = None):
+        """
+        Adds an entire stage or block of LAMMPS input commands and argument.
+
+        Args:
+            stage_commands:A dictionary containing LAMMPS commands and arguments
+                as key value pairs. Example: {'units': 'metal', 'atom_style': 'charge'}.
+                Alternatively, a list of strings containing both LAMMPS commands
+                and corresponding arguments can be provided. Example: ['units  metal',
+                'atom_style charge']
+            header: The header for the stage. The stage number will be used
+                as the header if not provided.
+            description: Add short description for this stage. This will
+                add inline with the stage header in the input
+        """
+
+        self.init_stage()
+
+        if header is None:
+            header = self.curr_stage_name
+
+        if description is not None:
+            self.add_comment(f"{header} : {description}", is_stage_header=True)
+        else:
+            self.add_comment(header, is_stage_header=True)  # type: ignore
+
+        self.add_commands(commands=stage_commands)
+
+    def generate_lammps_input(self):
+        """
+        Returns: LammpsInputFile object
+        """
+        return LammpsInputFile(self.input_settings)
```

Comment:
It's not clear to me what purpose this method serves. Isn't this basically returning a copy of the object itself? I think it can be removed.